src/fsharp/CompileOptions.fs

```diff
@@ -172,6 +172,32 @@ let DumpCompilerOptionBlocks blocks = List.iter dumpCompilerOptionBlock blocks
 let isSlashOpt (opt:string) = 
     opt.[0] = '/' && (opt.Length = 1 || not (opt.[1..].Contains "/"))
 
+module ResponseFile =
+
+    type ResponseFileData = ResponseFileLine list
+    and ResponseFileLine =
+        | CompilerOptionSpec of string
+        | Comment of string
+
+    let parseFile path : Choice<ResponseFileData,Exception> =
+        let parseLine (l: string) =
+            match l with
+            | s when String.IsNullOrWhiteSpace(s) -> None
+            | s when l.StartsWith("#") -> Some (ResponseFileLine.Comment (s.TrimStart('#')))
+            | s -> Some (ResponseFileLine.CompilerOptionSpec (s.Trim()))
+
+        try
+            use stream = FileSystem.FileStreamReadShim path
+            use reader = new System.IO.StreamReader(stream, System.Text.Encoding.UTF8)
```

Comment:
/cc @KevinRansom @dsyme  toughs about that? support only UTF8 or everything BOM support? i need to add a new `FileStreamReadShim` with detect?

>The detectEncodingFromByteOrderMarks parameter detects the encoding by looking at the first three bytes of the stream. It automatically recognizes UTF-8, little-endian Unicode, and big-endian Unicode text if the file starts with the appropriate byte order marks. Otherwise, the UTF8Encoding is used. See the Encoding.GetPreamble method for more information